lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Dictionary.java

```diff
@@ -251,34 +221,33 @@ public Dictionary(
   }
 
   /** Looks up Hunspell word forms from the dictionary */
-  IntsRef lookupWord(char word[], int offset, int length) {
-    return lookup(words, word, offset, length);
+  IntsRef lookupWord(char[] word, int length) {
+    return lookup(words, word, length);
   }
 
   // only for testing
-  IntsRef lookupPrefix(char word[], int offset, int length) {
-    return lookup(prefixes, word, offset, length);
+  IntsRef lookupPrefix(char[] word) {
+    return lookup(prefixes, word, word.length);
   }
 
   // only for testing
-  IntsRef lookupSuffix(char word[], int offset, int length) {
-    return lookup(suffixes, word, offset, length);
+  IntsRef lookupSuffix(char[] word) {
+    return lookup(suffixes, word, word.length);
   }
 
-  IntsRef lookup(FST<IntsRef> fst, char word[], int offset, int length) {
+  private IntsRef lookup(FST<IntsRef> fst, char[] word, int length) {
     if (fst == null) {
       return null;
     }
     final FST.BytesReader bytesReader = fst.getBytesReader();
-    final FST.Arc<IntsRef> arc = fst.getFirstArc(new FST.Arc<IntsRef>());
+    final FST.Arc<IntsRef> arc = fst.getFirstArc(new FST.Arc<>());
     // Accumulate output as we go
     final IntsRef NO_OUTPUT = fst.outputs.getNoOutput();
     IntsRef output = NO_OUTPUT;
 
-    int l = offset + length;
     try {
-      for (int i = offset, cp = 0; i < l; i += Character.charCount(cp)) {
```

Comment:
So this offset thing was never used? I this not useful to have keep it though? Utility methods could still exist (they'd use offset = 0) but it seems like some flexibility with offset here is fine.

Reply:
No one uses it now, and it's easy to add the offset back if needed. So far it doesn't seem very likely to happen.

Reply:
I'd rather keep it in if it's already there. Seems like it's just more complete this way (you point at a range of characters at the source buffer rather than require zero-based indexes). I'd even add a test that verifies it works by randomizing the offset somewhere...

Reply:
Having it here complicates the code and its refactoring a bit. That said, adding a test would remove IntelliJ's warning, so it's also a way to go. But why randomized? Passing `2` or something looks simple enough, and it doesn't seem that randomization would catch new bugs in this context.

Reply:
Passing a constant number gives you one test (which is fine). Passing a randomized number (0-X) gives you many tests over time. This seems stupid only until you find a bug deep down somewhere, hiding under a complex scenario of inputs. Maybe the offset is used to derive a different index, or an overflow occurs somewhere... I don't know but I've seen this too many times to call it redundant. ;)

Reply:
OK, having created a property-based test framework myself, I'm in a wrong position to argue about randomization usefulness. I'll try this, thanks!

Reply:
No pressure. :) I typically add known border cases and a randomized test on top of that. Fuzzifiers are great, really. Take a look at this bug which slipped through the cracks for so many years:
https://mail.openjdk.java.net/pipermail/javadoc-dev/2020-December/002255.html